warehouse/schema_test.go

```diff
@@ -3,225 +3,267 @@
+			existingDatatype: "string",
+			newDataType:      "float",
+			value:            1.501,
+			newColumnVal:     "1.501",
+			schemaChanged:    true,
+		},
+		{
+			name:             "should send string values if existing datatype is string, new datatype is datetime",
+			existingDatatype: "string",
+			newDataType:      "datetime",
+			value:            "2022-05-05T00:00:00.000Z",
+			newColumnVal:     "2022-05-05T00:00:00.000Z",
+			schemaChanged:    true,
+		},
+		{
+			name:             "should send string values if existing datatype is string, new datatype is string",
+			existingDatatype: "string",
+			newDataType:      "json",
+			value:            `{"json":true}`,
+			newColumnVal:     `{"json":true}`,
+			schemaChanged:    true,
+		},
+		{
+			name:             "should send json string values if existing datatype is json, new datatype is boolean",
+			existingDatatype: "json",
+			newDataType:      "boolean",
+			value:            false,
+			newColumnVal:     "false",
+			schemaChanged:    true,
+		},
+		{
+			name:             "should send json string values if existing datatype is jso, new datatype is int",
+			existingDatatype: "json",
+			newDataType:      "int",
+			value:            1,
+			newColumnVal:     "1",
+			schemaChanged:    true,
+		},
+		{
+			name:             "should send json string values if existing datatype is json, new datatype is float",
+			existingDatatype: "json",
+			newDataType:      "float",
+			value:            1.501,
+			newColumnVal:     "1.501",
+			schemaChanged:    true,
+		},
+		{
+			name:             "should send json string values if existing datatype is json, new datatype is json",
+			existingDatatype: "json",
+			newDataType:      "datetime",
+			value:            "2022-05-05T00:00:00.000Z",
+			newColumnVal:     "2022-05-05T00:00:00.000Z",
+			schemaChanged:    true,
+		},
+		{
+			name:             "should send json string values if existing datatype is json, new datatype is string",
+			existingDatatype: "json",
+			newDataType:      "string",
+			value:            "string value",
+			newColumnVal:     "string value",
+			schemaChanged:    true,
+		},
+		{
+			name:             "should send json string values if existing datatype is json, new datatype is array",
+			existingDatatype: "json",
+			newDataType:      "string",
+			value:            []interface{}{false, 1, "string value"},
+			newColumnVal:     "false",
+			schemaChanged:    true,
+		},
+		{
+			name:             "existing datatype is boolean, new datatype is int",
+			existingDatatype: "boolean",
+			newDataType:      "int",
+			value:            1,
+			schemaChanged:    false,
+		},
+		{
+			name:             "existing datatype is boolean, new datatype is float",
+			existingDatatype: "boolean",
+			newDataType:      "float",
+			value:            1.501,
+			schemaChanged:    false,
+		},
+		{
+			name:             "existing datatype is boolean, new datatype is string",
+			existingDatatype: "boolean",
```

Comment:
Shouldn't we get an error when we try to do something like that?

Reply:
Actually, it's an issue with the function signature we have, instead of with err we are handling it with boolean.
```
func HandleSchemaChange(existingDataType, currentDataType string, value any) (any, bool)
```

Reply:
We are now handling error as well.